consensus/src/state_computer.rs

```diff
@@ -85,7 +85,7 @@ impl StateComputer for ExecutionProxy {
                 .commit(committed_txns, reconfig_events)
                 .await
         ) {
-            error!("failed to notify state synchronizer: {:?}", e);
+            error!(error = "Failed to notify state synchronizer", "{:?}", e);
```

Comment:
Isn't `e` the error you want?

Reply:
what I understand is the `error` field would be indexed but the message won't. so I'd like to query the same type of error but the actual error message may vary

Reply:
There's a lot of extra data that is indexed with each log line. E.g. Log level, file name, line number, as well as the message itself. So what Greg is suggesting you do is:

```
error!(error = e, "Failed to notify state synchronizer");
```

Will result in structured data (that is indexed in Kibana) like:

```
{
  level: Error,
  file: "consensus/src/state_computer.rs",
  line: 88,
  message: "Failed to notify state synchronizer",
  data: {
    error: <Error formatted either by Serde/Debug/Display>,
  }
}
```

So everything is still indexable

Reply:
So I'm still able to filter by message = "Failed to notify state synchronizer"?

Reply:
You will, it'll to a text search, it just won't be quite as easy to filter, and won't be able to be used to be searched like that in visualizations.

Reply:
The main purpose of adding the extra fields is for indexing addition information, like identifiers, or in this case errors.  So, now I could have a visualization that says, show me a pie chart of all errors when `Failed to notify state synchronizer` and now I can see the percentage of errors by content

Reply:
You'd even be able to filter off the location, 'file:line-number' (which imo sounds like you're actually trying to do?)